packages/flutter/test/widgets/draggable_scrollable_sheet_test.dart

```diff
@@ -1746,4 +1746,30 @@ void main() {
       areCreateAndDispose,
     );
   });
+
+  testWidgets('DraggableScrollableSheet respects shouldCloseOnMinExtent', (WidgetTester tester) async {
+    final DraggableScrollableController controller = DraggableScrollableController();
+    DraggableScrollableNotification? receivedNotification;
+
+    Future<void> pumpWidgetAndFling() async {
+      await tester.pumpWidget(boilerplateWidget(
+        null,
+        controller: controller,
+        shouldCloseOnMinExtent: false,
+        onDraggableScrollableNotification: (DraggableScrollableNotification notification) {
+          receivedNotification = notification;
+          return false;
+        },
+      ));
+
+      await tester.flingFrom(const Offset(0, 325), const Offset(0, -325), 200);
+      await tester.pumpAndSettle();
+    }
+
+    await pumpWidgetAndFling();
+    expect(receivedNotification!.shouldCloseOnMinExtent, isFalse);
+    controller.jumpTo(0.5);
+    await pumpWidgetAndFling();
```

Comment:
I think we do still want to null out the notification here, to make sure that the `notification` isn't a leftover value.

It'd also be helpful to have a comment here to explain the rationale.

```suggestion
    expect(receivedNotification!.shouldCloseOnMinExtent, isFalse);

    receivedNotification = null;
    controller.jumpTo(0.5);

    // Construct the widget a second time, to ensure didUpdateWidget is called.
    await pumpWidgetAndFling();
```